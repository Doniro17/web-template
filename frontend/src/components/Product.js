import React from 'react';
import Button from './Button';
import API from '../utils/API';

export default function Product({ productId, imgSrc, name, description, price }) {
  async function addOrder() {
    await API.post(
      'orders',
      {
        products: [productId],
      },
      {
        withCredentials: true,
      },
    );
  }

  return (
    <div className="flex-item">
      <img className="product__image" src={imgSrc} alt={name} />
      <div className="item__data">
        <div className="product__title">{name}</div>
        <div className="product__description">{description}</div>
        <div className="product__price">{price}</div>
        <Button onClick={addOrder} type="button" className="new__button">
          Add to cart
        </Button>
      </div>
    </div>
  );
}
