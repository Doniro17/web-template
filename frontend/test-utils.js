import React from 'react'; // eslint-disable-line no-unused-vars

import { initFontAwesomeLibrary } from './src/utils';

import '@testing-library/jest-dom/extend-expect';

initFontAwesomeLibrary();

// re-export everything
export * from '@testing-library/react';
