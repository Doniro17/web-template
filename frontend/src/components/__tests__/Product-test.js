import { render } from 'test-utils';

import Product from '../Product';

const props = {
  key: 1,
  imgSrc: 'fakesource',
  name: 'testName',
  description: 'test description',
  price: 8800,
};
describe('Product', () => {
  it('renders correctly', () => {
    const { container } = render(<Product {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
