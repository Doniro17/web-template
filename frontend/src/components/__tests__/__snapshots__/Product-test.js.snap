// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product renders correctly 1`] = `
<div
  class="flex-item"
>
  <img
    alt="testName"
    class="product__image"
    src="fakesource"
  />
  <div
    class="item__data"
  >
    <div
      class="product__title"
    >
      testName
    </div>
    <div
      class="product__description"
    >
      test description
    </div>
    <div
      class="product__price"
    >
      8800
    </div>
    <button
      class="button"
      type="button"
    >
      Add to cart
    </button>
  </div>
</div>
`;
